jeta/archive/update.py

```diff
@@ -1483,46 +1264,94 @@ def _organize_data_for_append(large_sample, mdmap):
             _values[name].append(data['engineeringNumericValue'])
             # times are in milliseonds since unix epoch
             _times[name].append(data['observatoryTime'])
-        except:
+            # _counts[name] = _counts[name] + len(_times[name])
+        except Exception as err:
+            logger.error(f"Error: {err}")
             missing_ids.append(data['id'])
 
     return missing_ids
 
 
+def _sort_ingest_files_by_start_time(list_of_files=[]):
+    ingest_list = []
+
+    for file in list_of_files:
+        with h5py.File(file, 'r') as f:
+
+            tstart = f['samples']["data1"].attrs['dataStartTime'][0]/1000
+            tstop = f['samples'][f"data{len(f['samples'])}"].attrs['dataStopTime'][0]/1000
+            ingest_list.append(
+                {
+                    'filename': f.filename,
+                    'tstart': tstart,
+                    'tstop': tstop,
+                    'numPoints': f.attrs['/numPoints']
+                }
+            )
+
+    return sorted(ingest_list, key=lambda k: k['tstart'])
+
+
+def reset_storage():
+    global _values
+    _values = defaultdict(list)
+
+    global _times
+    _times = defaultdict(list)
+
+    global _counts
+    _counts = defaultdict(int)
+
+
 def process_ingest_files(files_to_process, tstart, tstop, chunk=4):
 
     processing_start_time = Time(Time.now(), format="datetime").iso
 
     file_processing_queue = deque(files_to_process)
+    original_queue_length = len(file_processing_queue)
+
+    processed_files = []
 
     while len(file_processing_queue) != 0:
 
-        large_sample = _allocate_large_sample(
-            len(file_processing_queue) * MAX_ROWS_PER_FILE
-        )
+        mdmap = {}
+        msids = []
+        offset = 0
+
+        reset_storage()
 
         if len(file_processing_queue) < chunk:
             chunk = len(file_processing_queue)
 
-        mdmap = {}
-        insert_offset = 0
-
         file_processing_chunk = [
             file_processing_queue.popleft()
             for i in range(chunk)
         ]
 
+        # Sum the number of points for a chunk to get pre-allocation value
+        num_points_in_chunk = sum([i['numPoints'] for i in file_processing_chunk])
+        num_points_in_chunk = num_points_in_chunk + (num_points_in_chunk * .01)
```

Comment:
num_points_in_chunk = int(1.01*num_points_in_chunk)

Reply:
Good idea! thank you for the suggestion :)